lib/y2_installbase.pm

```diff
@@ -255,6 +274,50 @@ sub select_all_patterns_by_menu {
     assert_screen 'inst-overview';
 }
 
+=head2 select_not_install_any_pattern 
+
+    select_not_install_any_pattern() 
+
+Being in the "select pattern" screen, performs steps to not install any
+patterns.
+=cut
+
+sub select_not_install_any_pattern {
+    my ($self) = @_;
+
+    # Ensure mouse on certain pattern then right click
+    assert_and_click("minimal-system", button => 'right');
+    assert_screen 'selection-menu';
+    # select action on all patterns
+    wait_screen_change { send_key 'a'; };
+    # confirm do not install
+    assert_and_click 'all-do-not-install';
+    save_screenshot;
+}
+
+=head2 select_current_visible_patterns 
+
+    select_current_visible_patterns()
+
+Being in the "select pattern" screen, performs steps to select visible
+patterns.
+=cut
+
+sub select_current_visible_patterns {
+    my ($self) = @_;
+    my @patterns = split(/,/, get_var('PATTERNS', ''));
+    # remove empty elements
+    @patterns = grep { $_ ne '' } @patterns;
+
+    for my $pattern (@patterns) {
+        # go to the top of the list before looking for the pattern
+        send_key "home";
+        # move the list of patterns down until the current pattern is found
+        send_key_until_needlematch "$pattern-pattern", "down", 80;
```

Comment:
why we cannot avoid iteration and just do `assert_and_click` ? that was the whole point to have a different needle, because the element is not selected, and the iteration one matches only with selected, because otherwise we could iterate the list with existing code without creating anything new...

Reply:
The purpose to do this iteration with down is change to next page for pattern list. As, the pattern list has very long list, it cannot show in page. If some patterns cannot be found at first page, we need this down key to move to next page to get the expected pattern.

Reply:
If the pattern can be found at first page, It will not send down key.

Reply:
we always expect the item to be found and first page, that is why we call it 'visible', the users of that method should know their exact location, that is why I think that we can put 'alt-home' or 'alt-end' at the level of test, so the user can use this shortcut (it is a bit of a hack, but if you have always the same patterns, works super fast).